ThatDefaultProject/Assets/ThatDefaultProject/Scripts/MonobehaviorExtensions.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace that
+{
+	public static class MonobehaviorExtensions
+	{
+		/// <summary>
+		/// if null, tries to get component, if still null, adds component. Returns component.
+		/// </summary>
+		/// <typeparam name="T"></typeparam>
+		/// <param name="mb"></param>
+		/// <param name="comp"></param>
+		/// <returns></returns>
+		public static T GetOrAddComponent<T>(this MonoBehaviour mb, T comp = null) where T : Component
+		{
+			if (comp != null)
+			{
+				return comp;
+			}
+			comp = mb.GetComponent<T>();
+			if (comp != null)
+			{
+				return comp;
+			}
+			return mb.gameObject.AddComponent<T>();
+		}
+		public static void SetActiveAll<T>(this IEnumerable<T> container, bool state) where T : Component
```

Comment:
I would rename this to "SetEnabledAll" so match the syntax Unity uses for Components comp.enabled = true/false

Reply:
Added 2 same functions that do the exact same but are named `SetEnableAll` in 7d6c2c26529ddbeb3f6a7e12edcf5a97b303d02f

Reply:
That works!